### PR TITLE
Handle empty repositories

### PR DESCRIPTION
**Pull request by pfeuffer**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -338,3 +338,5 @@ replace (
 	k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.29.6
 	k8s.io/sample-controller => k8s.io/sample-controller v0.29.6
 )
+
+replace github.com/scm-manager/goscm => /home/rene/development/cloudogu/goscm
```
Comment (zT-1337) [inline go.mod]: Reminder to remove this "replace"

File: applicationset/services/scm_provider/scm-manager_test.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
 )
 
-func scmManagerMockHandler(t *testing.T) func(http.ResponseWriter, *http.Request) {
+func scmManagerMockHandler(t *testing.T, emptyRepo bool) func(http.ResponseWriter, *http.Request) {
```
Comment (zT-1337) [inline applicationset/services/scm_provider/scm-manager_test.go]: Not a huge fan of the boolean argument and the switch case inside this MockHandler. I would prefer to have as little logic as possible within mocks and rather seperate it into different mock functions.

**Review by zT-1337 — CHANGES_REQUESTED** [review]

File: applicationset/services/scm_provider/scm-manager_test.go (hunk 2 above)
Comment (pfeuffer) [reply applicationset/services/scm_provider/scm-manager_test.go]: I can feel that. I've tried to make this better, but had no good idea. Maybe you can take another look...

Comment (zT-1337) [reply applicationset/services/scm_provider/scm-manager_test.go]: That is already an improvement. I would be fine merging this. But I will try to come up with an idea until monday.

**Pull request merged**